apps/meteor/app/lib/server/functions/saveUser/saveUser.ts

```diff
@@ -44,135 +47,148 @@ export type SaveUserData = {
 
 	joinDefaultChannels?: boolean;
 	sendWelcomeEmail?: boolean;
+
+	customFields?: Record<string, any>;
 };
 export type UpdateUserData = RequiredField<SaveUserData, '_id'>;
 export const isUpdateUserData = (params: SaveUserData): params is UpdateUserData => '_id' in params && !!params._id;
 
-export const saveUser = async function (userId: IUser['_id'], userData: SaveUserData) {
-	const oldUserData = userData._id && (await Users.findOneById(userData._id));
-	if (oldUserData && isUserFederated(oldUserData)) {
-		throw new Meteor.Error('Edit_Federated_User_Not_Allowed', 'Not possible to edit a federated user');
-	}
+const _saveUser = (session?: ClientSession) =>
+	async function (userId: IUser['_id'], userData: SaveUserData) {
+		const oldUserData = userData._id && (await Users.findOneById(userData._id));
```

Comment:
This one should receive the session as well.

Reply:
I was basing off of the mongodb docs. I read somewhere (could have been in another place) that if you know the document hasn't changed yet it's preferable to not pass the session. Since this is the first fetch of the user and happens before everything else, I though it was safe to not use the session here.

I can pass it anyway, don't think it'll do any harm.

Reply:
Couldn't this lead to issues in the microsservices mode? I believe that since this is not inside the transaction it will not guarantee this record to be consistent with the other ones.